### DISCUSSION TITLE
Use the DataToolkit.jl framework ?

### DISCUSSION THREAD

**Discussion by ederag** [brianguenter/OpticSim.jl#40] [Ideas]
About https://github.com/brianguenter/OpticSim.jl/pull/38#issuecomment-2322786318
> Download AGF files from various sources to a scratch-space (via Scratch.jl) rather than into the cloned repo itself

[DataToolkit.jl](https://github.com/tecosaur/DataToolkit.jl) might be relevant ?
It looks like a standard way to download, check and cache data,
with the possibility to extend this with a loader to get something that julia understands.
See the [documentation](https://tecosaur.github.io/DataToolkit.jl/main/#Declarative-data-management).

**Comment by brianguenter** [comment]
@ederag How did you get to the docs for [DataToolkit.jl](https://github.com/tecosaur/DataToolkit.jl)? The doc links on the repo are broken.

I am agnostic on the issue of Scratch.jl vs DataToolkit.jl - have no experience with either and no basis for choosing.

@bensetterholm do you have a preference?

**Reply by ederag** [reply]
> How did you get to the docs for [DataToolkit.jl](https://github.com/tecosaur/DataToolkit.jl)?

The badges links are broken (https://github.com/tecosaur/DataToolkit.jl/issues/46), so I used the first link in the README.

**Comment by bensetterholm** [comment]
I'm unfamiliar with this package, but it does look quite interesting! Let me play around with it for a bit.

From a very cursory look at the docs, I think this could be more of a both/and rather than an either/or question. Scratch gives you a consistent _place_ to store data whereas DataToolkit looks like a nice tool for _management_ of said data.
